packages/components/src/internal/actions.ts

```diff
@@ -1584,26 +1526,10 @@ function initEditorModel(model: QueryGridModel): void {
     updateEditorModel(newModel, {}, false);
 }
 
-export function clearSelection(modelId: string): void {
-    const model = getEditorModel(modelId);
-
-    if (model && (model.hasSelection() || model.hasFocus())) {
-        updateEditorModel(model, {
-            focusColIdx: -1,
-            focusRowIdx: -1,
-            selectedColIdx: -1,
-            selectedRowIdx: -1,
-            selectionCells: Set<string>(),
-        });
-    }
-}
-
-export function copyEvent(modelId: string, event: any): void {
-    const editorModel = getEditorModel(modelId);
-
+export function copyEvent(editorModel: EditorModel, insertColumns: List<QueryColumn>, event: any): void {
```

Comment:
These use immutable objects because insertColumns is coming from QueryInfo.getInsertColumns which returns an Immutable list. I can convert the functions to take an array and just call QueryInfo.getInsertColumns().toArray(). This would make it easier for us when we do eventually get around to removing Immutable.